pkg/lockfile/parse-apk-installed.go

```diff
@@ -0,0 +1,94 @@
+package lockfile
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const AlpineEcosystem Ecosystem = "Alpine"
+
+func groupApkPackageLines(scanner *bufio.Scanner) [][]string {
+	var groups [][]string
+	var group []string
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line == "" {
+			if len(group) > 0 {
+				groups = append(groups, group)
+			}
+			group = make([]string, 0)
+			continue
+		}
+		group = append(group, line)
+	}
+
+	if len(group) > 0 {
+		groups = append(groups, group)
+	}
+
+	return groups
+}
+
+func parseApkPackageGroup(group []string, pathToLockfile string) (PackageDetails, error) {
+	var pkg PackageDetails = PackageDetails{}
+
+	// File SPECS: https://wiki.alpinelinux.org/wiki/Apk_spec
+	for _, line := range group {
+		switch {
+		case strings.HasPrefix(line, "P:"):
+			pkg.Name = strings.TrimPrefix(line, "P:")
+			pkg.Ecosystem = AlpineEcosystem
+			pkg.CompareAs = AlpineEcosystem
+		case strings.HasPrefix(line, "V:"):
+			pkg.Version = strings.TrimPrefix(line, "V:")
+		case strings.HasPrefix(line, "c:"):
+			pkg.Commit = strings.TrimPrefix(line, "c:")
+		}
+	}
+
+	if pkg.Name == "" {
```

Comment:
We can simplify things a bit by moving this check out of this function - that way we can just always return `PackageDetails`, and the `ParseApkInstalled` function becomes responsible to deciding if we actually consider what we've parsed valid or not.

(You could also do that with `pkg.Version`, but I'd probably keep that in this function to be consistent with the similar function in the `yarn.lock` parser)

Reply:
Thank you @G-Rath , refactored as requested.
Also included commit to rename Yarn functions.